backends/occa/ceed-occa-elem-restriction.cpp

```diff
@@ -0,0 +1,435 @@
+        return NULL;
+      }
+
+      int ierr;
+      ElemRestriction *elemRestriction;
+
+      ierr = CeedElemRestrictionGetData(r, (void**) &elemRestriction);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetCeed(r, &elemRestriction->ceed);
+      CeedOccaFromChk(ierr);
+
+      return elemRestriction->setupFrom(r);
+    }
+
+    ElemRestriction* ElemRestriction::from(CeedOperatorField operatorField) {
+      int ierr;
+      CeedElemRestriction ceedElemRestriction;
+
+      ierr = CeedOperatorFieldGetElemRestriction(operatorField, &ceedElemRestriction);
+      CeedOccaFromChk(ierr);
+
+      return from(ceedElemRestriction);
+    }
+
+    ElemRestriction* ElemRestriction::setupFrom(CeedElemRestriction r) {
+      int ierr;
+
+      ierr = CeedElemRestrictionGetNumElements(r, &ceedElementCount);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetElementSize(r, &ceedElementSize);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetNumComponents(r, &ceedComponentCount);
+      CeedOccaFromChk(ierr);
+
+      ierr = CeedElemRestrictionGetLVectorSize(r, &ceedLVectorSize);
+      CeedOccaFromChk(ierr);
+
+      // Find what type of striding the restriction uses
+      bool isStrided = false;
+      bool hasBackendStrides = false;
+
+      ierr = CeedElemRestrictionGetStridedStatus(r, &isStrided);
+      CeedOccaFromChk(ierr);
+
+      if (isStrided) {
+        ierr = CeedElemRestrictionGetBackendStridesStatus(r, &hasBackendStrides);
+        CeedOccaFromChk(ierr);
+      }
+
+      if (isStrided) {
+        if (hasBackendStrides) {
+          ceedStrideType = FULLY_STRIDED;
+        } else {
+          ceedStrideType = NOT_STRIDED;
+        }
+      } else {
+        ceedStrideType = COMPONENT_STRIDED;
+      }
+
+      if (ceedStrideType == FULLY_STRIDED) {
+        CeedInt strides[3];
+
+        ierr = CeedElemRestrictionGetStrides(r, &strides);
+        CeedOccaFromChk(ierr);
+
+        ceedNodeStride      = strides[0];
+        ceedComponentStride = strides[1];
+        ceedElementStride   = strides[2];
+      } else if (ceedStrideType == COMPONENT_STRIDED) {
+        ierr = CeedElemRestrictionGetCompStride(r, &ceedComponentStride);
+        CeedOccaFromChk(ierr);
+      } else {
+        ceedNodeStride      = 1;
+        ceedComponentStride = ceedElementSize * ceedElementCount;
+        ceedElementStride   = ceedElementSize;
+      }
+
+      return this;
+    }
+
+    int ElemRestriction::apply(CeedTransposeMode rTransposeMode,
+                               Vector &u,
+                               Vector &v) {
+      const bool rIsTransposed = (rTransposeMode != CEED_NOTRANSPOSE);
+
+      ::occa::properties kernelProps;
+      kernelProps["defines/FULLY_STRIDED"]     = StrideType::FULLY_STRIDED;
+      kernelProps["defines/COMPONENT_STRIDED"] = StrideType::COMPONENT_STRIDED;
+      kernelProps["defines/NOT_STRIDED"]       = StrideType::NOT_STRIDED;
```

Comment:
Inside the libCEED nomenclature, I'd call these the following, based upon lines 260-279.

FULLY_STRIDED -> USER_STRIDES
COMPONENT_STRIDED -> NOT_STRIDED
NOT_STRIDED -> BACKEND_STRIDES